diff --git a/llvm/lib/Debuginfod/Debuginfod.cpp b/llvm/lib/Debuginfod/Debuginfod.cpp
--- a/llvm/lib/Debuginfod/Debuginfod.cpp	(revision 8e2cd28cd4ba46613a46467b0c91b1cabead26cd)
+++ b/llvm/lib/Debuginfod/Debuginfod.cpp	(date 1763274411135)
@@ -80,6 +80,9 @@
       StringRef(DebuginfodUrlsEnv)
           .split(DebuginfodUrls.value(), " ", -1, false);
     }
+    if (DebuginfodUrls->empty()) {
+      DebuginfodUrls->push_back("https://debuginfod.pwndbg.re");
+    }
     WriteGuard.unlock();
     ReadGuard.lock();
   }
