diff --git a/lldb/source/Target/Target.cpp b/lldb/source/Target/Target.cpp
--- a/lldb/source/Target/Target.cpp
+++ b/lldb/source/Target/Target.cpp
@@ -2616,8 +2616,9 @@ Target::GetScratchTypeSystemForLanguage(lldb::LanguageType language,
   if (!m_valid)
     return llvm::createStringError("Invalid Target");

-  if (language == eLanguageTypeMipsAssembler // GNU AS and LLVM use it for all
-                                             // assembly code
-      || language == eLanguageTypeUnknown) {
+  if (language == eLanguageTypeMipsAssembler  // GNU AS and LLVM use it for all
+                                              // assembly code
+      || language == eLanguageTypeAssembly    // DWARFv6 generic assembly type
+      || language == eLanguageTypeUnknown) {
     LanguageSet languages_for_expressions =
         Language::GetLanguagesSupportingTypeSystemsForExpressions();

