Index: gdb/utils.c
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/gdb/utils.c b/gdb/utils.c
--- a/gdb/utils.c	(revision 631a49c452a4a456dd9889d172541ea789f8bcae)
+++ b/gdb/utils.c	(date 1766818215698)
@@ -1706,11 +1706,12 @@
 		 don't increment chars_printed here.  */

 	      size_t style_len;
-	      ui_file_style style;
+          ui_file_style style = m_applied_style;
 	      if (style.parse (linebuffer, &style_len)
 		  && style_len <= skip_bytes)
 		{
 		  this->emit_style_escape (style);
+          m_applied_style = style;

 		  linebuffer += style_len;
 		  skip_bytes -= style_len;
