--- a/gdb/python/python-config.py	(revision b61f17e6951ff630bc9090b0e82ec9b551497f92)
+++ b/gdb/python/python-config.py	(date 1763714357526)
@@ -1,6 +1,15 @@
 # Program to fetch python compilation parameters.
 # Copied from python-config of the 2.7 release.
 
+# In this script, we should only use the following to retrieve configuration values:
+# - `sysconfig.get_config_var`
+# - `sysconfig.get_platform`
+# This is because certain variables may return invalid data during cross-compilation, for example:
+# - sys.prefix -> Use sysconfig.get_config_var("prefix") instead.
+# - sysconfig.get_path("include") -> Don't use, it may return paths for native python, not our target python
+# - os.name -> Use sysconfig.get_platform() for platform detection.
+
+
 import getopt
 import os
 import sys
@@ -26,7 +35,7 @@
 
 pyver = sysconfig.get_config_var("VERSION")
 getvar = sysconfig.get_config_var
-abiflags = getattr(sys, "abiflags", "")
+abiflags = getvar("ABIFLAGS")
 
 opt_flags = [flag for (flag, val) in opts]
 
@@ -49,15 +58,14 @@
 
 for opt in opt_flags:
     if opt == "--prefix":
-        print(to_unix_path(os.path.normpath(sys.prefix)))
+        print(to_unix_path(os.path.normpath(getvar("prefix"))))
 
     elif opt == "--exec-prefix":
-        print(to_unix_path(os.path.normpath(sys.exec_prefix)))
+        print(to_unix_path(os.path.normpath(getvar("exec_prefix"))))
 
     elif opt in ("--includes", "--cflags"):
         flags = [
-            "-I" + sysconfig.get_path("include"),
-            "-I" + sysconfig.get_path("platinclude"),
+            "-I" + getvar("INCLUDEPY"),
         ]
         if opt == "--cflags":
             flags.extend(getvar("CFLAGS").split())
@@ -65,7 +73,7 @@
 
     elif opt in ("--libs", "--ldflags"):
         libs = ["-lpython" + pyver + abiflags]
-        if getvar("LIBS") is not None:
+        if getvar("LIBS") is not None and pyver != "3.10":
             libs.extend(getvar("LIBS").split())
         if getvar("SYSLIBS") is not None:
             libs.extend(getvar("SYSLIBS").split())
@@ -76,7 +84,7 @@
                 if getvar("LIBPL") is not None:
                     libs.insert(0, "-L" + getvar("LIBPL"))
                 elif os.name == "nt":
-                    libs.insert(0, "-L" + os.path.normpath(sys.prefix) + "/libs")
+                    libs.insert(0, "-L" + os.path.normpath(getvar("prefix")) + "/libs")
             if getvar("LINKFORSHARED") is not None:
                 libs.extend(getvar("LINKFORSHARED").split())
         print(to_unix_path(" ".join(libs)))